File: scala-advent-of-code/src/test/scala/advent/twenty_fifteen/Day05DoesntHeHaveInternElvesForThisTest.scala
package advent.twenty_fifteen

import org.scalatest.{FlatSpec, Matchers}

class Day05DoesntHeHaveInternElvesForThisTest extends FlatSpec with Matchers {
  "2015 Day 3 - Input" should "calculate answers" in {
    val dp = new Day05DoesntHeHaveInternElvesForThis("/twenty_fifteen/Day05-DoesntHeHaveInternElvesForThis-input.txt")
    dp.part1Answer should be(258)
    dp.part2Answer should be(53)
  }

  "2015 Day 3 - Example #1" should "calculate answers" in {
    val dp = new Day05DoesntHeHaveInternElvesForThis("/twenty_fifteen/Day05-DoesntHeHaveInternElvesForThis-example#1.txt")
    dp.part1Answer should be(2)
    dp.part2Answer should be(2)
  }

}

File: java-advent-of-code/src/test/java/advent/twenty_fifteen/Day05DoesntHeHaveInternElvesForThisTest.java
package advent.twenty_fifteen;

import advent.common.DailyProblem;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class Day05DoesntHeHaveInternElvesForThisTest {
    @Test
    public void checkReal() {
        DailyProblem<Integer, Integer> dp = new Day05DoesntHeHaveInternElvesForThis("/twenty_fifteen/Day05-DoesntHeHaveInternElvesForThis-input.txt");

        assertEquals(258, dp.getPart1Answer().intValue());
        assertEquals(53, dp.getPart2Answer().intValue());
    }

    @Test
    public void checkExample1() {
        DailyProblem<Integer, Integer> dp = new Day05DoesntHeHaveInternElvesForThis("/twenty_fifteen/Day05-DoesntHeHaveInternElvesForThis-example#1.txt");

        assertEquals(2, dp.getPart1Answer().intValue());
        assertEquals(2, dp.getPart2Answer().intValue());
    }
}